webauthn/helpers/parse_authentication_credential_json.py

```diff
@@ -0,0 +1,21 @@
+from typing import Callable
+from pydantic import ValidationError
+
+from .exceptions import InvalidAuthenticationResponse
+from .structs import PYDANTIC_V2, AuthenticationCredential
+
+
+def parse_authentication_credential_json(json_val: str) -> AuthenticationCredential:
+    if PYDANTIC_V2:
+        parsing_method: Callable = AuthenticationCredential.model_validate_json  # type: ignore[attr-defined]
+    else:  # assuming V1
+        parsing_method = AuthenticationCredential.parse_raw
+
+    try:
+        authentication_credential = parsing_method(json_val)
+    except ValidationError as exc:
+        raise InvalidAuthenticationResponse(
+            "Unable to parse an authentication credential from JSON data"
+        ) from exc
```

Comment:
I'm not familiar with this `from exc` syntax like this in an `except` - does it surface anything from the original exception so users have a chance to understand what specifically was bad?

Reply:
Exactly. In this way you're explicitly declaring that exception `exc` (an instance of `ValidationError`) was the cause of your new `InvalidAuthenticationResponse` exception.

If you try/except `InvalidAuthenticationResponse` when calling `parse_authentication_credential_json` there will be a `__cause__` attribute that you can inspect if you need to know more about what caused it.

It's called explicit exception chaining. All the info [here](https://peps.python.org/pep-0344/#explicit-exception-chaining).